src/vivarium_public_health/results/disability.py

```diff
@@ -64,19 +68,54 @@ def __init__(self) -> None:
     def setup(self, builder: Builder) -> None:
         self.step_size = pd.Timedelta(days=builder.configuration.time.step_size)
         self.disability_weight = self.get_disability_weight_pipeline(builder)
-        self.causes_of_disease = [
-            cause
-            for cause in builder.components.get_components_by_type(
-                tuple(self.disease_classes)
+        self.set_causes_of_disability(builder)
+
+    def set_causes_of_disability(self, builder: Builder) -> None:
+        """Set the causes of disability to be observed by removing any excluded
+        via the model spec from the list of all disability class causes. We implement
+        exclusions here because disabilities are unique in that they are not
+        registered stratifications and so cannot be excluded during the stratification
+        call like other categories.
+        """
+        causes_of_disability = list(
+            builder.components.get_components_by_type(tuple(self.disability_classes))
+        )
+        # Convert to SimpleCause instances and add on all_causes
+        causes_of_disability = [
+            SimpleCause.create_from_disease_state(cause) for cause in causes_of_disability
+        ] + [SimpleCause("all_causes", "all_causes", "cause")]
+
+        excluded_causes = (
+            builder.configuration.stratification.excluded_categories.to_dict().get(
+                "disability", []
+            )
+        )
+
+        # Handle exclusions that don't exist in the list of causes
+        cause_names = [cause.state_id for cause in causes_of_disability]
+        unknown_exclusions = set(excluded_causes) - set(cause_names)
+        if len(unknown_exclusions) > 0:
```

Comment:
But we use `unknown_exclusions` in the error message too. Would you prefer `if unknown_exclusions:` instead of a length check? (I prefer being explicit but don't feel strongly)